src/sqlfluff/core/rules/base.py

```diff
@@ -372,7 +391,7 @@ class FunctionalRuleContext:
     def __init__(self, context: RuleContext):
         self.context = context
 
-    @cached_property
+    @property
```

Comment:
Eliminated caching for most fields now that the same `RuleContext` instance is used throughout processing of a rule. Caching would be wrong in this case (or we'd have to keep clearing the cache between segments).